Apple/__main__.py

```diff
@@ -0,0 +1,988 @@
+
+
+def build(context: argparse.Namespace, host: str | None = None) -> None:
+    """The implementation of the "build" command."""
+    if host is None:
+        host = context.host
+
+    if context.clean:
+        clean(context, host)
+
+    if host in {"all", "build"}:
+        for step in [
+            configure_build_python,
+            make_build_python,
+        ]:
+            step(context)
+
+    if host == "build":
+        hosts = []
+    elif host in {"all", "hosts"}:
+        hosts = all_host_triples(context.platform)
+    else:
+        hosts = [host]
+
+    for step_host in hosts:
+        for step in [
+            configure_host_python,
+            make_host_python,
+        ]:
+            step(context, host=step_host)
+
+    if host in {"all", "hosts"}:
+        package(context)
+
+
+def test(context: argparse.Namespace, host: str | None = None) -> None:
+    """The implementation of the "test" command."""
+    if host is None:
+        host = context.host
+
+    if context.clean:
+        clean(context, "test")
+
+    with group(f"Test {'XCframework' if host in {'all', 'hosts'} else host}"):
+        timestamp = str(time.time_ns())[:-6]
+        testbed_dir = (
+            CROSS_BUILD_DIR / f"{context.platform}-testbed.{timestamp}"
+        )
+        if host in {"all", "hosts"}:
+            framework_path = (
+                CROSS_BUILD_DIR / context.platform / "Python.xcframework"
+            )
+        else:
+            build_arch = platform.machine()
+            host_arch = host.split("-")[0]
+
+            if not host.endswith("-simulator"):
+                print("Skipping test suite non-simulator build.")
+                return
+            elif build_arch != host_arch:
+                print(
+                    f"Skipping test suite for an {host_arch} build "
+                    f"on an {build_arch} machine."
+                )
+                return
+            else:
+                framework_path = (
+                    CROSS_BUILD_DIR
+                    / host
+                    / f"Apple/{context.platform}"
+                    / f"Frameworks/{apple_multiarch(host)}"
+                )
+
+        run(
+            [
+                sys.executable,
+                "Apple/testbed",
+                "clone",
+                "--platform",
+                context.platform,
+                "--framework",
+                framework_path,
+                testbed_dir,
+            ]
+        )
+
+        run(
+            [
+                sys.executable,
+                testbed_dir,
+                "run",
+                "--verbose",
+            ]
+            + (
+                ["--simulator", str(context.simulator)]
+                if context.simulator
+                else []
+            )
+            + [
+                "--",
+                "test",
+                "-uall",
+                "--single-process",
+                "--rerun",
```

Comment:
This could match the options used by Android in #138649:

```suggestion
                "--fast-ci",
                "--single-process",
                "--no-randomize",
```